src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4916,6 +4916,19 @@ syntax switch
                     }
                 }
             }
+            else
+            {
+                // normally we delay visiting the lambda until we can visit it along with its conversion.
+                // since we can't visit its conversion here, or it doesn't have one, we dig back in and
+                // visit the lambda here to ensure all nodes have nullability info for public API
```

Comment:
One part I'm having trouble connecting the dots on is how this fixes the performance issue here. Does this for instance get caches populated sooner?

Reply:
It just means that we visit a lambda once in an analysis pass instead of visiting it twice, which has exponential runtime when lambdas are nested as mentioned in the description.